src/ast/nodes/ObjectExpression.ts

```diff
@@ -159,29 +202,45 @@ export default class ObjectExpression extends NodeBase {
 		predicateFunction: SomeReturnExpressionCallback,
 		options: ExecutionPathOptions
 	): boolean {
-		if (path.length === 0) return true;
-		const subPath = path[0];
-		if (path.length === 1 && typeof subPath === 'string' && objectMembers[subPath]) {
-			return predicateFunction(options)(objectMembers[subPath].returns);
+		const key = path[0];
+		if (
+			path.length === 0 ||
+			this.hasUnknownReassignedProperty ||
+			(typeof key === 'string' && this.reassignedPaths[key])
+		)
+			return true;
+
+		if (path.length === 1 && typeof key === 'string' && objectMembers[key]) {
+			return predicateFunction(options, objectMembers[key].returns);
```

Comment:
I wasn't able to come up with any test that checks this behaviour. Any ideas? Would be nice to capture as much of this as possible through coverage.

Reply:
You are very right. I added some tests around these lines and also discovered a bug while doing so...